From 1481a4ce4ceea1ae98ef1cadfe1b4e0d75e41e85 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@tq-group.com>
Date: Fri, 28 Apr 2023 14:07:39 +0200
Subject: [PATCH] iMX8M: print_fit_hab: fix output for symlinked inputs

Without -L, stat will return the length of the symlinked path instead of
the size of the target file for symlinks.

Signed-off-by: Matthias Schiffer <matthias.schiffer@tq-group.com>
---
 iMX8M/print_fit_hab.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/iMX8M/print_fit_hab.sh b/iMX8M/print_fit_hab.sh
index 06fca5f..bbf2090 100755
--- a/iMX8M/print_fit_hab.sh
+++ b/iMX8M/print_fit_hab.sh
@@ -33,7 +33,7 @@ else
 	let uboot_sign_off=$((fit_off - 0x8000 - ivt_off + $FIT_DATA_POS))
 fi
 
-let uboot_size=$(stat --printf="%s" $BL33)
+let uboot_size=$(stat --dereference --printf="%s" $BL33)
 let uboot_load_addr=0x40200000
 
 let last_sign_off=$(((uboot_sign_off + uboot_size + 3) & ~3))
@@ -50,7 +50,7 @@ for dtname in $*
 do
 	if [ ${cnt} != 0 ]
 	then
-		let fdt${cnt}_size=$(stat --printf="%s" $dtname)
+		let fdt${cnt}_size=$(stat --dereference --printf="%s" $dtname)
 
 		let fdt${cnt}_sign_off=$((last_sign_off))
 		let fdt${cnt}_load_addr=$((last_load_addr))
@@ -71,13 +71,13 @@ done
 
 let atf_sign_off=$((last_sign_off))
 let atf_load_addr=$ATF_LOAD_ADDR
-let atf_size=$(stat --printf="%s" $BL31)
+let atf_size=$(stat --dereference --printf="%s" $BL31)
 
 if [ ! -f $BL32 ]; then
 	let tee_size=0x0
 	let tee_sign_off=$((atf_sign_off + atf_size))
 else
-	let tee_size=$(stat --printf="%s" $BL32)
+	let tee_size=$(stat --dereference --printf="%s" $BL32)
 
 	let tee_sign_off=$(((atf_sign_off + atf_size + 3) & ~3))
 	let tee_load_addr=$TEE_LOAD_ADDR
-- 
2.25.1

